crates/bevy_app/src/app.rs

```diff
@@ -80,8 +80,8 @@ pub struct App {
     sub_apps: HashMap<AppLabelId, SubApp>,
     plugin_registry: Vec<Box<dyn Plugin>>,
     plugin_name_added: HashSet<String>,
-    /// A private marker to prevent incorrect calls to `App::run()` from `Plugin::build()`
-    is_building_plugin: bool,
+    /// A private counter to prevent incorrect calls to `App::run()` from `Plugin::build()`
+    building_plugin_counter: usize,
```

Comment:
```suggestion
    building_plugin_depth: usize,
```

Counter implies that it's how many have been built or are building at the same time? Probably worth using depth here for clarity.

Reply:
I agree, `building_plugin_depth` is the better name.